src/debuginfo.cpp

```diff
@@ -861,10 +880,22 @@ void jl_getFunctionInfo(char **name, size_t *line, char **filename,
         if (*line == (size_t) -1) {
             *line = prev.Loc.getLine();
         }
+
+        DILexicalBlockFile locscope = DILexicalBlockFile(prev.Loc.getScope((*it).second.func->getContext()));
+        jl_copy_str(filename, (char*)(uintptr_t)locscope.getFilename().data());
```

Comment:
> Agreed. We seem to have gotten away with it so far.

Interesting thought, but we've also had a number of over-reads on strings (sometimes turning into segfaults) from the usage of `.data()` instead of `.c_str()`. For StringRef, use `.str().c_str()`, or just modify `jl_copy_str` to take the length as an argument since the information should have always been present anyways.